### PR TITLE
[INJIMOB-1588] mdoc mock support

### PR DESCRIPTION
**Pull request by KiruthikaJeyashankar**
TODO
- Post mock-certify-plugin's library version is updated, make required changes here

### CODE REVIEW

File: certify-service/src/main/resources/application-local.properties [hunk 1]
```diff
@@ -241,6 +238,13 @@ mosip.certify.integration.scan-base-package=io.mosip.certify.mock.integration
 mosip.certify.integration.audit-plugin=LoggerAuditService
 mosip.certify.integration.vci-plugin=MockVCIssuancePlugin
 mosip.certify.mock.vciplugin.verification-method=${mosip.certify.authn.jwk-set-uri}
+#mosip.certify.mock.authenticator.get-identity-url="https://api.dev1.mosip.net/v1/mock-identity-system/identity"
+mosip.certify.mock.authenticator.get-identity-url="http://localhost:8082/v1/mock-identity-system/identity"
+mosip.certify.mock.vciplugin.p12.filename=mock-issuer-details.p12
+#TODO: get the secret for password onboarded
+mosip.certify.mock.vciplugin.p12.password=${mosip.certify.mock.vciplugin.keystore.password}
```
Comment (KiruthikaJeyashankar) [inline certify-service/src/main/resources/application-local.properties:245]: Please let me know if any name changes are required for the secret "mosip.certify.mock.vciplugin.keystore.password"

File: certify-service/src/main/java/io/mosip/certify/services/VCIssuanceServiceImpl.java [hunk 2]
```diff
@@ -75,22 +73,30 @@ public class VCIssuanceServiceImpl implements VCIssuanceService {
...
             throw new NotAuthenticatedException();
 
         String scopeClaim = (String) parsedAccessToken.getClaims().getOrDefault("scope", "");
         CredentialMetadata credentialMetadata = null;
-        for(String scope : scopeClaim.split(Constants.SPACE)) {
-            Optional<CredentialMetadata> result = getScopeCredentialMapping(scope);
-            if(result.isPresent()) {
-                credentialMetadata = result.get(); //considering only first credential scope
-                break;
+        if(!credentialRequest.getFormat().equals(VCFormats.MSO_MDOC)){
```
Comment (vharsh) [inline certify-service/src/main/java/io/mosip/certify/services/VCIssuanceServiceImpl.java:87]: TODO(vharsh): discuss with @jainhitesh9998 why the scope validation not required for mDoc VCs.

**Review by vharsh — COMMENTED** [review]
minor comments, looks good overall 👍🏼

File: certify-core/src/main/java/io/mosip/certify/core/validators/MsoMdocCredentialRequestValidator.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+import io.mosip.certify.core.dto.CredentialRequest;
+
+public class MsoMdocCredentialRequestValidator implements CredentialRequestValidator {
+    @Override
+    public boolean isValidCheck(CredentialRequest credentialRequest) {
+        if (credentialRequest.getDoctype() == null || credentialRequest.getDoctype().isBlank()) {
+            return false;
+        }
+        if (credentialRequest.getClaims() == null || credentialRequest.getClaims().isEmpty())
```
[the remark this answers is not in the document]
Comment (KiruthikaJeyashankar) [reply certify-core/src/main/java/io/mosip/certify/core/validators/MsoMdocCredentialRequestValidator.java:11]: claims is a map, and doctype is string for. this reason empty check is alone done in claims

File: certify-service/src/main/java/io/mosip/certify/services/VCIssuanceServiceImpl.java [hunk 4]
```diff
@@ -120,14 +123,15 @@ private VCResult<?> getVerifiableCredential(CredentialRequest credentialRequest,
         parsedAccessToken.getClaims().put("accessTokenHash", parsedAccessToken.getAccessTokenHash());
         VCRequestDto vcRequestDto = new VCRequestDto();
         vcRequestDto.setFormat(credentialRequest.getFormat());
-        vcRequestDto.setContext(credentialRequest.getCredential_definition().getContext());
-        vcRequestDto.setType(credentialRequest.getCredential_definition().getType());
-        vcRequestDto.setCredentialSubject(credentialRequest.getCredential_definition().getCredentialSubject());
+
```
Comment (vharsh) [inline certify-service/src/main/java/io/mosip/certify/services/VCIssuanceServiceImpl.java:126]: nitpick(optional): lines 132-134 & 143-146 seem exactly similar can they be moved back outside the switch-case & the try-catch?

This should come as a Sonar error later, hence asked.

File: certify-service/src/main/java/io/mosip/certify/services/VCIssuanceServiceImpl.java [hunk 5]
```diff
@@ -164,14 +177,15 @@ private CredentialResponse<?> getCredentialResponse(String format, VCResult<?> v
...
+            case VCFormats.MSO_MDOC :
                 CredentialResponse<String> jsonResponse = new CredentialResponse<>();
                 jsonResponse.setCredential((String)vcResult.getCredential());
                 return jsonResponse;
         }
         throw new CertifyException(ErrorConstants.UNSUPPORTED_VC_FORMAT);
     }
 
-    private Optional<CredentialMetadata>  getScopeCredentialMapping(String scope) {
+    private Optional<CredentialMetadata>  getScopeCredentialMapping(String scope, String format) {
```
Comment (vharsh) [inline certify-service/src/main/java/io/mosip/certify/services/VCIssuanceServiceImpl.java:188]: TODO(vharsh): if the `scope` check is exclusively required for LDP_VCs this should get moved to `LdpVcCredentialRequestValidator`. This makes the code cleaner by placing _like with like things_

wdyt @jainhitesh9998, is it okay to share the `issuerMetadata` with the validator impl assuming similar requirements to prop up on Mdoc credential issuance ?

**Review by vharsh — COMMENTED** [review]
adding some more minor discussion points

File: certify-service/src/main/java/io/mosip/certify/services/VCIssuanceServiceImpl.java (hunk 4 above)
Comment (KiruthikaJeyashankar) [reply certify-service/src/main/java/io/mosip/certify/services/VCIssuanceServiceImpl.java:126]: mso_mdoc format VC does not have credential_definition as part of credential request, for this reason we cannot move it outside switch, as its not a common piece of code for different VC formats, we can try extracting a separate method for this (May be like `populateVCRequestDTOWithCredentialDefinitionDetails`)

File: certify-service/src/main/java/io/mosip/certify/services/VCIssuanceServiceImpl.java (hunk 5 above)
Comment (KiruthikaJeyashankar) [reply certify-service/src/main/java/io/mosip/certify/services/VCIssuanceServiceImpl.java:188]: Adding my point of view, I think a design change in inji-certify to handle vc issuance for different VC formats would be able to tackle this issue

File: certify-service/src/main/java/io/mosip/certify/services/VCIssuanceServiceImpl.java (hunk 2 above)
Comment (KiruthikaJeyashankar) [reply certify-service/src/main/java/io/mosip/certify/services/VCIssuanceServiceImpl.java]: Scope validation is now added for mdoc VCs as well

**Pull request closed without merging**